### PR TITLE
README: initial notes about Resolvers and Horizons

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,3 +7,68 @@
 [godoc-badge]: https://pkg.go.dev/badge/darvaza.org/penne.svg
 [goreport]: https://goreportcard.com/report/darvaza.org/penne
 [goreport-badge]: https://goreportcard.com/badge/darvaza.org/penne
+
+_Penne_ is a config-driven pipeline oriented DNS resolver that allows complex
+flows simply defined.
```
Comment (karasz) [inline README.md:12]: that allows complex workflows to be defined in a simple way.

File: README.md [hunk 2]
```diff
@@ -7,3 +7,68 @@
...
+[core]: https://pkg.go.dev/darvaza.org/core
+[resolver]: https://pkg.go.dev/darvaza.org/resolver
+[sidecar]: https://pkg.go.dev/darvaza.org/sidecar
+[slog]: https://pkg.go.dev/darvaza.org/slog
+
+## Horizons
+
+A _Horizon_ is a named set of network patterns (aka `CIDR`) that can optionally choose a custom `Resolver`
+and can annotate requests before passing them to the next _Horizon_ on a chain.
+
```
Comment (karasz) [inline README.md:25]: I would mention here the `split` word as maybe people are more familiar with the `split horizon` sintagm

File: README.md [hunk 3]
```diff
@@ -7,3 +7,68 @@
...
+_Resolvers_ have three operation modes:
+
+* _Iterative_ goes to the root servers and iterates through authoritative
+  servers until the answer is found.
+* _Forwarder_  connects to a specific server to get the answer, optionally
+  allowing recursion to be performed remotely.
+* and _Chained_, where requests are passed to the _Next_ resolver, optionally modified.
+
+_Resolvers_ act as middleware, optionally restricted to specific domains (suffixes).
```
Comment (karasz) [inline README.md:40]: _Resolvers_ act as middlewares

File: README.md [hunk 4]
```diff
@@ -7,3 +7,68 @@
...
+
+_Resolvers_ can also be configured to discard `AAAA` entries and execute request rewrites.
```
Comment (karasz) [inline README.md:42]: various entries (like `AAAA` for example)

**Review by karasz — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (amery) [reply README.md:32]: check now

**Comment by amery** [comment]
rebased and amended as requested.

**Review by karasz — APPROVED** [review]
\o/

**Pull request merged**